### ISSUE TITLE
Better Support for Linked & Unlinked References in Namespaces

### ISSUE THREAD

**Issue by jacklaing** [Roam-Research/issues#323]
**Problem**

Namespacing seems like it would be a useful way to differentiate page types but, as it stands, attributes (e.g. Type:: [[Person]]) are more effective for the following reasons:
- Using the attribute method links to a [[Person]] page that acts like an index for all pages of that type
- Because it creates a link, the attribute method ensures that the page will be included if you do a query with [[Person]]
-  Using the attribute method doesn't break the unlinked references workflow. People would not write "Person/John Smith" unless they were intentionally linking to the namespaced page already, so the unlinked references for [[Person/John Smith]] should return all references to "John Smith".

**Proposed Solution**

My proposed solution, which I think would also address these issues ([234](https://github.com/Roam-Research/issues/issues/234), [18](https://github.com/Roam-Research/issues/issues/18)), is:
- Typing [[Person/John Smith]] creates a [[Person]] page and a [[Person/John Smith]] page, and links them together. This means namespacing automatically creates an index under the Person page and queries which ask for [[Person]] will return the [[Person/John Smith]] page.
- In the Person/John Smith page, “Person” is ignored for the purposes of Unlinked References. Let's say we have [[Person/John Smith]] and [[Topic/John Smith]], "John Smith" would appear as an unlinked reference in both, until it was linked to either.

**Comment by jacklaing** [comment]
Some further thoughts:
- Linked references to [[John Smith]], without any namespaces, should be treated as Unlinked References to [[Person/John Smith]] and [[Topic/John Smith]], in addition to unlinked instances of "John Smith". This will be important especially in multiplayer Roams, where people might turn John Smith into a page but forget to specify a namespace.
- As for clicking on the link. Clicking the "Person/" section would take you to the [[Person]] index page, whereas clicking on the "/John Smith" section would take you to the [[Person/John Smith]] page.

**Comment by mmower** [comment]
I'm not sure about the specific solutions here, but I support the general idea that namespace references need fleshing out.

**Comment by Itshalffull** [comment]
Oh, that makes sense! I agree with your solution.

**Comment by mmower** [comment]
Having had a chance to review Jack's proposed solutions I think I agree with them.

**Comment by irace** [comment]
I agree with these proposals as well! Would be a big improvement.

**Comment by ErikPlachta** [comment]
Agree that this would be an excellent improvement

**Comment by phhq** [comment]
I agree with the approach also, but there is another option to consider, which would help along the way. Add at least one new option to the toggle for namespaces, such that nested tags or pages in any namespace (everything before the final "/") would remain hidden. Examples:

Today, user can choose to view [[Presidents/Thomas Jefferson]] as:

1. President/Thomas Jefferson (usual color text for links)
2. P/Thomas Jefferson (in different color text)
3. Thomas Jefferson (different color text)

That's all fine. But if we actively tag [[President]] and [[Thomas Jefferson]] inside of [[[[President]]/[[Thomas Jefferson]]]], the good thing is now both [[President]] then [[Thomas Jefferson]] will be linked from instances of the other. However, the visualization options 2 and 3 are no longer possible. 

So I propose a 4th option, which is to treat nested page links as links when selecting 2 or 3. Meaning, if toggled, even [[[[President]]/[[Thomas Jefferson]]]] would still be viewable as
 
3. Thomas Jefferson (different color text)

This may be more helpful than it seems at first glance, as it would actually make better use of nested pages without changing much about the internal working of roam or namespaces, while keeping the page refs usable and meaningful.

**Comment by trbielec** [comment]
I have avoided using Namespaces for a variety of reasons including the ones mentioned above.

Bumping this. But also thank you @jacklaing for suggesting using "`Type:: [[Person]]`" in the meantime
